File: App/src/Adaptive.ReactiveTrader.Client/UI/Shell/ShellViewModel.cs
using Adaptive.ReactiveTrader.Client.UI.Blotter;
using Adaptive.ReactiveTrader.Client.UI.SpotTiles;
using PropertyChanged;

namespace Adaptive.ReactiveTrader.Client.UI.Shell
{
    [ImplementPropertyChanged]
    public class ShellViewModel : ViewModelBase, IShellViewModel
    {
        public ISpotTilesViewModel SpotTiles { get; private set; }
        public IBlotterViewModel Blotter { get; private set; }

        public ShellViewModel(ISpotTilesViewModel spotTiles, IBlotterViewModel blotter)
        {
            SpotTiles = spotTiles;
            Blotter = blotter;
        }
    }
}

File: src/Adaptive.ReactiveTrader.Client.JS/UI/Shell/ShellViewModel.ts
class ShellViewModel {
    spotTiles: ISpotTilesViewModel;
    blotter: IBlotterViewModel;
    connectivityStatus: IConnectivityStatusViewModel;

    constructor(
        spotTiles: ISpotTilesViewModel,
        blotter: IBlotterViewModel,
        connectivityStatus: IConnectivityStatusViewModel) {
        this.spotTiles = spotTiles;
        this.blotter = blotter;
        this.connectivityStatus = connectivityStatus;
    }
} 